pwiz_tools/Skyline/Model/Proteome/ProteinGroupMetadata.cs

```diff
@@ -0,0 +1,216 @@
+            proteinMetadataList[matchingProteinIndex] = singleProteinMetadata;
+            return new ProteinGroupMetadata(Name, Description, proteinMetadataList);
+        }
+
+        public override ProteinMetadata ChangeWebSearchInfo(WebSearchInfo webSearchInfo)
+        {
+            return new ProteinGroupMetadata(this, webSearchInfo);
+        }
+
+        public override ProteinMetadata ClearWebSearchInfo()
+        {
+            // sometimes all you really want is to initialize
+            return new ProteinGroupMetadata(this, WebSearchInfo.EMPTY);
+        }
+
+        public override ProteinMetadata SetWebSearchCompleted()
+        {
+            return new ProteinGroupMetadata(this, WebSearchInfo.SetSearchCompleted());
+        }
+
+        public override ProteinMetadata SetWebSearchTerm(WebSearchTerm search)
+        {
+            return new ProteinGroupMetadata(this, WebSearchInfo.SetSearchTerm(search));
+        }
+
+        /// <summary>
+        /// returns a copy of this, with any blanks filled in
+        /// using the members of source
+        /// </summary>
+        /// <param name="source">the source to merge from</param>
+        /// <returns>a copy of this merged with source, with this winning when fields conflict</returns>
+        public override ProteinMetadata Merge(ProteinMetadata source)
+        {
+            if (source == null)
+                return this;
+            if (source is ProteinGroupMetadata sourceGroup)
+            {
+                return new ProteinGroupMetadata(String.IsNullOrEmpty(Name) ? sourceGroup.Name : Name,
+                    String.IsNullOrEmpty(Description) ? sourceGroup.Description : Description,
+                    sourceGroup.ProteinMetadata);
+            }
+
+            return new ProteinGroupMetadata(String.IsNullOrEmpty(Name) ? source.Name : Name,
+                String.IsNullOrEmpty(Description) ? source.Description : Description,
+                ProteinMetadata);
+            //throw new InvalidOperationException("cannot merge ProteinMetadata into ProteinGroupMetadata");
+        }
+
+        private ProteinGroupMetadata() : base(null, null)
+        {
+        }
+
+        private ProteinGroupMetadata(ProteinGroupMetadata other, WebSearchInfo webSearchInfo = null) : base(other.Name, other.Description)
+        {
+            webSearchInfo ??= other.ProteinMetadata.First().WebSearchInfo;
+            ProteinMetadata = ImmutableList<ProteinMetadata>.ValueOf(other.ProteinMetadata);
+            //ProteinMetadata[0] = ProteinMetadata[0].ChangeWebSearchInfo(webSearchInfo);
+            /*    Name = ProteinMetadata.All(p => p.Name == null)
+                ? null
+                : string.Join(GROUP_SEPARATOR, ProteinMetadata.OrderBy(p => p.).Select(p => p.Name));*/
+        }
+
+        public ProteinGroupMetadata(string name, string description, IList<ProteinMetadata> proteinMetadata) : base(name, description)
```

Comment:
Prefer to see the constructors closer to the top just after privates and statics.